src/common/tracer.h

```diff
@@ -0,0 +1,80 @@
+#ifndef TRACER_H_
+#define TRACER_H_
+
+#define SIGNED_RIGHT_SHIFT_IS 1
+#define ARITHMETIC_RIGHT_SHIFT 1
+
+#include<cstring>
+#include <arpa/inet.h>
+#include <yaml-cpp/yaml.h>
+#include<atomic>
+
+#include <jaegertracing/Tracer.h>
+#include <jaegertracing/net/IPAddress.h>
+#include <jaegertracing/net/Socket.h>
+
+
+typedef std::unique_ptr<opentracing::Span> Span;
+
+struct Jaeger_Tracer{
+    Jaeger_Tracer(){}
+    ~Jaeger_Tracer();
+  std::shared_ptr<opentracing::v3::Tracer> tracer = NULL;
+};
+
+//will be used throughout ceph to create spans
+extern Jaeger_Tracer tracer;
+extern std::atomic<bool> jaeger_initialized;
+
+
+static inline void init_tracer(const char* tracerName,const char* filePath){
```

Comment:
why do you implement in header. can be moved to cc?

Reply:
this will also allow you to remove many of the includes here, and move them in the cc file only

Reply:
My main objective of having tracer.cc is to define the declared extern variable, should I just move those rgw_common.cc?